src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/EntityCreateConnectionRequestController.php

```diff
@@ -48,28 +48,66 @@ private function validateServiceIsAllowed(int $serviceId, string $manageId, stri
 
     /**
      * @Method({"GET", "POST"})
-     * @Route("/entity/create-connection-request/{environment}/{manageId}/{serviceId}", name="entity_published_create_connection_request")
+     * @Route("/entity/create-connection-request-from-entity/{environment}/{manageId}/{serviceId}", name="entity_published_create_connection_request_from_entity")
      * @throws \Exception
      */
-    public function changeRequestAction(
+    public function connectionRequestFromEntityAction(
         Request $request,
         int $serviceId,
         string $manageId,
         string $environment
     ): Response {
+        $manageEntity = $this->entityService->getManageEntityById($manageId, $environment);
+        $applicant = $this->authorizationService->getContact();
 
         $this->validateServiceIsAllowed($serviceId, $manageId, $environment);
 
-        $command = new CreateConnectionRequestCommand();
+        $command = new CreateConnectionRequestCommand($manageEntity, $applicant);
         $form = $this->createForm(ConnectionRequestContainerType::class, $command);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $this->commandBus->handle($command);
-            $this->addFlash('info', 'entity.create_connection_request.edit.flash.success');
-            // Simply return to entity list, no entity was saved
-            if ($this->isGranted('ROLE_ADMINISTRATOR')) {
-                return $this->redirectToRoute('service_admin_overview', ['serviceId' => $serviceId]);
+            if (!$this->isCancelAction($form)) {
+                $this->commandBus->handle($command);
+            }
+            // Show Thank you page entity is published
+            return $this->redirectToRoute('entity_published_production');
+        }
+
+        return $this->render(
+            '@Dashboard/EntityPublished/createConnectionRequest.html.twig',
+            ['form' => $form->createView()]
+        );
+    }
+
+    /**
+     * @Method({"GET", "POST"})
+     * @Route("/entity/create-connection-request-from-overview/{environment}/{manageId}/{serviceId}", name="entity_published_create_connection_request_from_overview")
+     * @throws \Exception
+     */
+    public function connectionRequestFromOverviewAction(
```

Comment:
Why have two controller actions that do practically the same? `connectionRequestFromOverviewAction` vs `connectionRequestFromEntityAction`.

If your aim is to have two different routes for the same action, then just simply define multiple routes on the same action.

Reply:
Good to know that different routes can be applied to one action. The action is a bit different.